jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/MultiException.java

```diff
@@ -13,36 +13,149 @@
+ *       {
+ *         try
+ *         {
+ *             task.run();
+ *         }
+ *         catch(Throwable t)
+ *         {
+ *             multiException.add(t)
+ *         }
+ *       }
+ *     }
+ *     multiException.ifExceptionalThrow();
+ * </pre>
+ * <p>In some cases, the alternative style can be further simplified:
+ * <pre>
+ *     MultiException.Builder multiException = new MultiException.Builder();
+ *     manyThings.forEach(multiException::callAndCatch);
+ *     multiException.ifExceptionalThrow();
+ * </pre>
+ *
+ * </p>
  *
- * Allows multiple exceptions to be thrown as a single exception.
  *
- * The MultiException itself should not be thrown instead one of the
- * ifExceptionThrow* methods should be called instead.
- * TODO: Remove entirely in favor of Error lists with {@link TypeUtil#withSuppressed(Throwable, List)}?
- * TODO: Deprecate this now (if we do above)?
  */
 @SuppressWarnings("serial")
 public class MultiException extends Exception
 {
+    public static class Builder
+    {
+        private Throwable _multiException;
+
+        public Builder add(Throwable t)
+        {
+            _multiException = MultiException.add(_multiException, t);
+            return this;
+        }
+
+        public Builder ifExceptionThrow() throws Exception
+        {
+            MultiException.ifExceptionThrow(_multiException);
+            return this;
+        }
+
+        public Builder ifExceptionThrowRuntime() throws Exception
+        {
+            MultiException.ifExceptionThrowRuntime(_multiException);
+            return this;
+        }
+
+        public Builder callAndCatch(Invocable.Callable task)
+        {
+            try
+            {
+                if (task != null)
+                    task.call();
+            }
+            catch (Throwable t)
+            {
+                add(t);
+            }
+            return this;
+        }
+
+
+    }
+
+    public static Throwable add(Throwable multi, Throwable t)
+    {
+        if (multi == null)
+            return t;
+        if (!TypeUtil.isAssociated(multi, t))
+            multi.addSuppressed(t);
+        return multi;
+    }
+
+    public static void ifExceptionThrow(Throwable multi)
+        throws Exception
+    {
+        if (multi == null)
+            return;
+        if (multi instanceof Exception exception)
+            throw exception;
+        if (multi instanceof Error error)
+            throw error;
+        throw new RuntimeException(multi);
+    }
+
+    public static void ifExceptionThrowRuntime(Throwable multi)
+        throws RuntimeException
+    {
+        if (multi == null)
+            return;
+        if (multi instanceof RuntimeException runtimeException)
+            throw runtimeException;
+        if (multi instanceof Error error)
+            throw error;
+        throw new RuntimeException(multi);
+    }
+
     private static final String DEFAULT_MESSAGE = "Multiple exceptions";
 
+    @Deprecated
```

Comment:
If we are keeping `MultiException`, why are we deprecating?
Just remove the fields/methods, or set them private to disallow use.
Heck set the class itself to be final too.

Reply:
This current version keeps MultiException around in more or less it's current form as a deprecated backwards compatible API.
But if we are happy to break backward compatibility in major version (I am), then renaming to `ExceptionUtil` makes sense.... we can also move the new `TypeUtil.isAssociated` over to `ExceptionUtil`